Storage/Storage/CoreData/CoreDataManager.swift

```diff
@@ -351,7 +378,7 @@ extension CoreDataManagerError: CustomStringConvertible {
 ///
 extension CoreDataManager {
     /// Helper type to support handling async operations by keeping track of their states
-    class AsyncOperation: Operation {
+    class AsyncOperation: Operation, @unchecked Sendable {
```

Comment:
This change adds the concurrency suggestion in Xcode 16.

Reply:
Since this doesn't trigger a warning or error in < Xcode 16, should be still part of this PR, or move it to the migration project when does happen? 

I lack context on this bit, andI have no strong opinion about it, but less changes should keep the core data threading project tighter. We also have the chance to test later if there are further implications for conforming to `@unchecked Sendable` and conform to the compiler that `AsyncOperation` is okay to be used in concurrent contexts without going through all the checks.

Reply:
`@unchecked Sendable` is used for `Operation`, which is the parent class of `AsyncOperation` and `AsyncBlockOperation`. That's why Xcode requires restating inherited conformance - it's not a declaration for threading behavior. I kept this as-is for this PR.